aws-iot-domainconfiguration/src/main/java/com/amazonaws/iot/domainconfiguration/DeleteHandler.java

```diff
@@ -0,0 +1,99 @@
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class DeleteHandler extends BaseHandler<CallbackContext> {
+    private static final String DELETE_OPERATION = "DeleteDomainConfiguration";
+    private static final String UPDATE_OPERATION = "UpdateDomainConfiguration";
+
+    private IotClient iotClient;
+
+    public DeleteHandler() {
+        iotClient = ClientBuilder.getClient();
+    }
+
+    public DeleteHandler(IotClient iotClient) {
+        this.iotClient = iotClient;
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        final ResourceModel model = request.getDesiredResourceState();
+        final String domainConfigName = model.getDomainConfigurationName();
+
+        String operation = DELETE_OPERATION;
+
+        final DeleteDomainConfigurationRequest deleteRequest = DeleteDomainConfigurationRequest.builder()
+                .domainConfigurationName(domainConfigName)
+                .build();
+        boolean previouslyDisabled = callbackContext != null && callbackContext.isDomainConfigurationDisabled();
+        try {
+            if(!previouslyDisabled &&
+                    !StringUtils.equals(DomainConfigurationStatus.DISABLED.toString(), model.getDomainConfigurationStatus())) {
+                operation = UPDATE_OPERATION;
+                final UpdateDomainConfigurationRequest updateRequest = UpdateDomainConfigurationRequest.builder()
+                        .domainConfigurationName(model.getDomainConfigurationName())
+                        .authorizerConfig(ResourceUtil.getSdkAuthorizerConfig(model))
+                        .domainConfigurationStatus(DomainConfigurationStatus.DISABLED.toString())
+                        .removeAuthorizerConfig(false)
+                        .build();
+                UpdateDomainConfigurationResponse response = proxy.injectCredentialsAndInvokeV2(updateRequest, iotClient::updateDomainConfiguration);
+                logger.log(String.format("%s [%s] set as %s before deleting",
+                        ResourceModel.TYPE_NAME, model.getDomainConfigurationName(), DomainConfigurationStatus.DISABLED));
+
+                return ProgressEvent.defaultInProgressHandler(
+                        CallbackContext.builder().domainConfigurationDisabled(true).build(),
+                        ResourceUtil.DELAY_CONSTANT,
+                        ResourceModel.builder()
+                                .domainConfigurationArn(response.domainConfigurationArn())
+                                .domainConfigurationName(response.domainConfigurationName())
+                                .build());
+            }
+
+            proxy.injectCredentialsAndInvokeV2(deleteRequest, iotClient::deleteDomainConfiguration);
```

Comment:
sorry, I wasn't following the discussion, what did we decide on deleting domain configuration after 7day wait period?

Reply:
The delete call will fail with an InvalidRequestException - which we send it back to the with the message as to why the delete failed. The user is responsible for waiting for 7 days. This follows the pattern for other resources - The user must put the resource in a delible state before deleting it

Reply:
Yes it does, but the status reason for the delete failed will contain a message like this, so they will be aware as to why the specific delete request failed.
`Invalid request provided: Request: DeleteDomainConfigurationRequest(DomainConfigurationName=SampleDomainConfig12345) Message: AWS Managed Domain Configuration must be disabled for at least 7 days before it can be deleted (Service: Iot, Status Code: 400, Request ID: 12345678-1234-1234-1234-1234567890ab, Extended Request ID: null)`
So the user should remove the deletion of the domain configuration from the update request and trigger it again for it to succeed.

Reply:
thanks for the clarification.